sklearn/manifold/t_sne.py

```diff
@@ -707,7 +710,7 @@ def _fit(self, X, skip_num_points=0):
             raise ValueError("n_iter should be at least 200")
 
         if self.metric == "precomputed":
-            if self.init == 'pca':
+            if not isinstance(self.init, np.ndarray) and self.init == 'pca':
```

Comment:
I didn't, that's why! I'll go back to using 'string_types' and get it from there.